recipes/libvncserver/all/conanfile.py

```diff
@@ -0,0 +1,161 @@
+        if self.options.with_systemd:
+            self.requires("pkgconf/[>=2 <3]")
+            self.requires("libsystemd/253.10")
+        if self.options.with_libgcrypt:
+            #self.requires("libgcrypt/1.8.4")
+            print("Not yet implemented")
+        if self.options.with_ffmpeg:
+            self.requires("ffmpeg/6.0")
+        if self.options.with_sasl:
+            self.requires("cyrus-sasl/2.1.27")
+        if self.options.with_xcb:
+            self.requires("xorg/system")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.variables["WITH_TESTS"] = False
+        tc.variables["WITH_EXAMPLES"] = False
+
+        tc.variables["WITH_ZLIB"] = self.options.with_zlib
+        tc.variables["WITH_LZO"] = self.options.with_lzo
+        tc.variables["WITH_JPEG"] = self.options.with_libjpeg
+        tc.variables["WITH_PNG"] = self.options.with_libpng
+        tc.variables["WITH_SDL"] = self.options.with_sdl
+        tc.variables["WITH_GTK"] = self.options.with_gtk
+        tc.variables["WITH_LIBSSH2"] = self.options.with_libssh2
+        tc.variables["WITH_OPENSSL"] = self.options.with_openssl
+        tc.variables["WITH_SYSTEMD"] = self.options.with_systemd
+        tc.variables["WITH_GCRYPT"] = self.options.with_libgcrypt
+        tc.variables["WITH_FFMPEG"] = self.options.with_ffmpeg
+        tc.variables["WITH_SASL"] = self.options.with_sasl
+        tc.variables["WITH_XCB"] = self.options.with_xcb
+
+        tc.generate()
+
+    def build(self):
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_find_mode", "both")
+        self.cpp_info.set_property("cmake_file_name", "LibVncServer")
+        self.cpp_info.set_property("cmake_target_name", "libvncserver::libvncserver")
+        #self.cpp_info.libs = ["libvncserver", "libvncclient"]
+        self.cpp_info.libs = ['vncserver']
+        #self.cpp_info.libs = collect_libs(self)
+        #print(self.cpp_info.libs)
+
+        self.cpp_info.components["libvncclient"].set_property("cmake_target_name", "LibVncClient")
+        self.cpp_info.components["libvncclient"].libs = ['vncclient']
+        self.cpp_info.components["libvncclient"].set_property("cmake_target_name", "libvncserver::libvncclient")
+
```

Comment:
```suggestion
        self.cpp_info.components["server"].set_property("pkg_config_name", "libvncserver")
        self.cpp_info.components["client"].set_property("pkg_config_name", "libvncclient")

```
It generates pc files too: https://github.com/LibVNC/libvncserver/blob/LibVNCServer-0.9.14/CMakeLists.txt#L808